src/components/Upload/UploadItem.tsx

```diff
@@ -0,0 +1,104 @@
+import {
+  Box,
+  IconButton,
+  LinearProgress,
+  makeStyles,
+  Typography,
+} from '@material-ui/core';
+import { Cancel as CancelIcon } from '@material-ui/icons';
+import React, { FC } from 'react';
+import { UploadFile } from './Reducer';
+
+const useStyles = makeStyles(({ palette, spacing }) => ({
+  container: {
+    padding: spacing(1),
+    '&:not(:last-of-type)': {
+      borderBottom: `1px solid ${palette.divider}`,
+    },
+  },
+  fileName: {
+    width: '18ch',
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+    whiteSpace: 'nowrap',
+  },
+  progressBarBox: {
+    marginTop: spacing(2.5),
+  },
+  percentage: {
+    minWidth: '2ch',
+  },
+  statusText: {
+    fontSize: '0.65rem',
+  },
+  clearButton: {
+    color: palette.error.main,
+  },
+}));
+
+interface UploadItemProps {
+  file: UploadFile;
+  onClear: () => void;
+}
+
+export const UploadItem: FC<UploadItemProps> = (props) => {
+  const { file, onClear } = props;
+  const { error, fileName, percentCompleted, uploadId } = file;
+  const classes = useStyles();
+
+  const progressLabel = error
+    ? error.message
+    : !uploadId
+    ? 'Initializing'
+    : percentCompleted === 100
+    ? 'Completed'
+    : 'Uploading';
+
+  return (
+    <Box
```

Comment:
I still don't want to use `Box` since it creates new css with every mount. I think `<Grid>` should get you pretty close to what you are already doing here.